src/main/java/com/kszydlo1/ticket_booking_app/views/ScreeningsView.java

```diff
@@ -0,0 +1,77 @@
+package com.kszydlo1.ticket_booking_app.views;
+
+import com.kszydlo1.ticket_booking_app.Constants;
+import com.kszydlo1.ticket_booking_app.model.database.Screening;
+import com.kszydlo1.ticket_booking_app.model.requests.ScreeningsPeriodRequest;
+import com.kszydlo1.ticket_booking_app.model.responses.ScreeningsPeriodResponse;
+import com.kszydlo1.ticket_booking_app.repository.ScreeningRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Vector;
+import java.util.stream.Collectors;
+
+@RestController
+public class ScreeningsView {
+    private class StartEndDateException extends Exception {};
```

Comment:
1. Jaka jest konsekwencja tego, że ta klasa nie jest statyczna?
2. Jak ktoś ma złapać ten wyjątek jeśli jest prywatny?

Reply:
1. Można tworzyć instancje kontrolerów (co istotnie nie ma zbyt dużego sensu).
2. Zakładałem, że te wyjatki mają być łapane tylko i wyłącznie wewnątrz klasy.

Reply:
Ad 1. Co innego mnie tu boli. Chodzi o to, że jeśli klasa wewnętrzna nie jest statyczna, to trzyma w sobie referencję do klasy zewnętrznej. Czyli Twój wyjątek trzyma cały kontroler ;)